code/modules/mob/living/silicon/robot/robot_items.dm

```diff
@@ -137,4 +137,126 @@
 	name = "mobility module"
 	desc = "By retracting limbs and tucking in its head, a combat android can roll at high speeds."
 	icon = 'icons/obj/decals.dmi'
-	icon_state = "shock"
\ No newline at end of file
+	icon_state = "shock"
+
+#define MODE_WALL 0
+#define MODE_DOOR 1
+
+/obj/item/weapon/inflatable_dispenser
+	name = "inflatables dispenser"
+	desc = "A hand-held device which allows rapid deployment and removal of inflatable structures."
+	icon = 'icons/obj/storage.dmi'
+	icon_state = "inf_deployer"
+	w_class = W_CLASS_MEDIUM
+
+	var/list/stored_walls = list()
+	var/list/stored_doors = list()
+	var/max_walls = 4
+	var/max_doors = 3
+	var/list/allowed_types = list(/obj/item/inflatable/wall, /obj/item/inflatable/door)
+	var/mode = MODE_WALL
+
+/obj/item/weapon/inflatable_dispenser/New()
+	..()
+	for(var/i = 0 to max(max_walls,max_doors))
+		if(stored_walls.len < max_walls)
+			stored_walls += new /obj/item/inflatable/wall(src)
+		if(stored_doors.len < max_doors)
+			stored_doors += new /obj/item/inflatable/door(src)
+
+/obj/item/weapon/inflatable_dispenser/Destroy()
+	stored_walls = null
+	stored_doors = null
+	..()
+
+/obj/item/weapon/inflatable_dispenser/robot
+	w_class = W_CLASS_HUGE
```

Comment:
Hows this work?  I can't get it to enter my "hand" slots.

https://a.pomf.cat/neajfj.gif